releasetool/github.py

```diff
@@ -21,22 +22,57 @@
 
 _GITHUB_ROOT: str = "https://api.github.com"
 _GITHUB_UI_ROOT: str = "https://github.com"
+_MAGIC_GITHUB_PROXY_ROOT: str = "http://magic-github-proxy.endpoints.devrel-prod.cloud.goog"
 
 
 class GitHub:
-    def __init__(self, token: str) -> None:
+    def __init__(self, token: str, use_proxy: bool = False) -> None:
         self.session: requests.Session = requests.Session()
+        self.GITHUB_ROOT = _GITHUB_ROOT
         self.session.headers.update(
             {
                 "Accept": "application/vnd.github.v3+json",
                 "Authorization": f"Bearer {token}",
             }
         )
 
+        if use_proxy:
+            self.GITHUB_ROOT = _MAGIC_GITHUB_PROXY_ROOT
+
+            # To use the proxy, we need an api key for the magic github proxy.
```

Comment:
Will you pluck this out to its own function, maybe `_find_devrel_api_key`?

Reply:
Will do. :+1: